xlive/H2MOD/Modules/Stats/StatsHandler.cpp

```diff
@@ -0,0 +1,686 @@
+	curl_easy_setopt(curl, CURLOPT_MIMEPOST, form);
+	result = curl_easy_perform(curl);
+	if(result != CURLE_OK)
+	{
+		LOG_ERROR_GAME(L"[H2MOD]::playlistVerified failed to execute curl");
+		//LOG_ERROR_GAME(curl_easy_strerror(result));
+		return -1;
+	}
+	else
+	{
+		int response_code;
+		curl_easy_getinfo(curl, CURLINFO_RESPONSE_CODE, &response_code);
+		if (verbose)
+			LOG_INFO_GAME(std::to_string(response_code));
+		return response_code;
+	}
+	return -1;
+}
+
+
+
+int StatsHandler::verifyPlaylist()
+{
+	if (verbose)
+		LOG_INFO_GAME(L"Verifying Playlist..");
+	std::string http_request_body = "https://www.halo2pc.com/test-pages/CartoStat/API/get.php?Type=PlaylistCheck&Playlist_Checksum=";
+	wchar_t* playlist_file = getPlaylistFile();
+	if (verbose)
+		LOG_INFO_GAME(L"[H2MOD] playlistVerified loaded playlist is {0}", playlist_file);
+
+	std::string checksum = getChecksum();
+	if (checksum == "")
+	{
+		LOG_ERROR_GAME(L"[H2MOD] playlistVerified failed to Hash Playlist file");
+		return -1;
+	}
+	else
+	{
+		if (verbose)
+			LOG_INFO_GAME(checksum);	
+		http_request_body.append(checksum);
+		if (verbose)
+			LOG_INFO_GAME(http_request_body);
+		char* Response;
+		CURL *curl;
+		CURLcode curlResult;
+		CURLcode global_init = curl_global_init(CURL_GLOBAL_ALL);
+		if(global_init != CURLE_OK)
+		{
+			LOG_INFO_GAME(L"[H2MOD]::playlistVerified failed to init curl");
+			return -1;
+		}
+		curl = curl_easy_init();
+		if(curl)
+		{
+			//Set the URL for the GET
+			curl_easy_setopt(curl, CURLOPT_URL, http_request_body.c_str());
+			curl_easy_setopt(curl, CURLOPT_SSL_VERIFYHOST, 0L);
+			curl_easy_setopt(curl, CURLOPT_SSL_VERIFYPEER, 0L);
+			curlResult = curl_easy_perform(curl);
+			if(curlResult != CURLE_OK)
+			{
+				LOG_ERROR_GAME(L"[H2MOD]::playlistVerified failed to execute curl");
+				//LOG_ERROR_GAME(curl_easy_strerror(curlResult));
+				return -1;
+			} else
+			{
+				int response_code;
+				curl_easy_getinfo(curl, CURLINFO_RESPONSE_CODE, &response_code);
+				if (verbose)
+					LOG_INFO_GAME(std::to_string(response_code));
+				return response_code;
+			}
+			curl_easy_cleanup(curl);
+		}
+	}
+	return -1;
+}
+
+
+int StatsHandler::uploadStats(char* filepath)
+{
+	CURL *curl;
+	CURLcode result;
+	curl_mime *form = NULL;
+	curl_mimepart *field = NULL;
+	struct curl_slist *headerlist = NULL;
+	static const char buf[] = "Expect:";
+	curl_global_init(CURL_GLOBAL_ALL);
+	curl = curl_easy_init();
+	if (!curl)
+	{
+		LOG_ERROR_GAME(L"[H2MOD] uploadStats failed to init curl");
+		return -1;
+	}
+	form = curl_mime_init(curl);
+	field = curl_mime_addpart(form);
+	curl_mime_name(field, "file");
+	curl_mime_filedata(field, filepath);
+	field = curl_mime_addpart(form);
+	curl_mime_name(field, "Type");
+	curl_mime_data(field, "GameStats", CURL_ZERO_TERMINATED);
+	headerlist = curl_slist_append(headerlist, buf);
+
+	curl_easy_setopt(curl, CURLOPT_URL, "https://www.halo2pc.com/test-pages/CartoStat/API/post.php");
```

Comment:
your implementation is not secure

Reply:
Added JWT Token Authentication and made it that servers have to manually be enabled for collection.